pkg/plugin/processor/builtin/impl/avro/schemaregistry/fake.go

```diff
@@ -94,10 +94,10 @@ const (
 	errorCodeSchemaNotFound  = 40403
 )
 
-// fakeRegistry is a simple fake registry meant to be used in tests. It stores
+// FakeRegistry is a simple fake registry meant to be used in tests. It stores
 // schemas in memory and supports only the basic functionality needed in our
 // tests and supported by our client.
-type fakeRegistry struct {
+type FakeRegistry struct {
 	schemas            []sr.SubjectSchema
```

Comment:
So the idea is to completely rely on Apicurio? I was actually talking about changing this struct so it becomes an actual schema registry. All it's missing is just a way to permanently store the schemas, that's what I mean by giving it access to a [`database.DB`](https://github.com/ConduitIO/conduit/blob/bf18b8c4649f0c418ba5bc93f92cfbdacd2ae09a/pkg/foundation/database/database.go#L31) object and let it store the schemas there.

What worries me is that if we don't have a schema registry in here, then Conduit will have a hard dependency on Apicurio (well, depending on the connectors used in the pipeline) and won't be a completely standalone tool anymore, which is one of the biggest appeals of Conduit. I'm not saying that we shouldn't use an external registry, we should, that's how schemas can be shared across Conduit instances. But I'm trying to prevent the external registry becoming a hard dependency, things should ideally still work if there is none.

Reply:
Leaving the no-op (in memory) implementation around for development purposes is a good idea, we can do that. :+1: 

As for storing the schemas in the Conduit DB, I think that too is a good idea, but we should discuss it as a separate item, so that expectations are correctly, especially given that we plan a Conduit schema registry long term anyway.